File: javascript/stringToInteger.js
function myAtoi(s) {
	// return Math.max(-(2**31),Math.min(2**31-1,parseInt(s)||0))

	const str = s.trim().split('');
	let res = 0;
	let sign = 1;
	let i = 0;
	if (str[0] == '-' || str[0] == '+') {
		sign = str[0] === '-' ? -1 : 1;
		i++;
	}
	while (i < str.length && !isNaN(parseInt(str[i]))) {
		res = res * 10 + parseInt(str[i]);
		i++;
	}
	return sign * res < -2147483648
		? -2147483648
		: Math.min(2147483647, sign * res);
};

File: python/string_to_integer.py
# convert string to intger
# input: '   -448 word', '+98s', '0047b'
def myAtoi(s: str) -> int:
        s = s.strip(' ')
        if len(s) == 0:
            return 0

        res = 0
        sign = 1
        if s[0] == '-' or s[0] == '+':
            sign = -1 if s[0] == '-' else 1
            s= s[1:]
        
        i = 0
        while len(s) > i and s[i].isdigit():
            # ans += s[i]
            res = res*10 + int(s[i])
            i += 1
        
        return max(-2147483648, min(sign*res,2147483647))